velox/dwio/parquet/tests/ParquetTableScanTest.cpp

```diff
@@ -25,64 +25,279 @@ using namespace facebook::velox;
+    rowTypes_ = rowTypes;
+    createDuckDbTable({data});
   }
 
   std::string getExampleFilePath(const std::string& fileName) {
     return facebook::velox::test::getDataFilePath(
         "velox/dwio/parquet/tests", "examples/" + fileName);
   }
 
+ private:
   std::shared_ptr<connector::hive::HiveConnectorSplit> makeSplit(
       const std::string& filePath) {
     auto split = makeHiveConnectorSplit(filePath);
     split->fileFormat = dwio::common::FileFormat::PARQUET;
     return split;
   }
+
+  std::shared_ptr<const RowType> rowTypes_;
+  std::vector<std::shared_ptr<connector::ConnectorSplit>> splits_;
 };
 
 TEST_F(ParquetTableScanTest, basic) {
-  auto data = makeRowVector(
-      {"a", "b"},
-      {
-          makeFlatVector<int64_t>(20, [](auto row) { return row + 1; }),
-          makeFlatVector<double>(20, [](auto row) { return row + 1; }),
-      });
-  createDuckDbTable({data});
+  loadData(
+      getExampleFilePath("sample.parquet"),
+      ROW({"a", "b"}, {BIGINT(), DOUBLE()}),
+      makeRowVector(
+          {"a", "b"},
+          {
+              makeFlatVector<int64_t>(20, [](auto row) { return row + 1; }),
+              makeFlatVector<double>(20, [](auto row) { return row + 1; }),
+          }));
 
-  auto split = makeSplit(getExampleFilePath("sample.parquet"));
+  // Plain select
+  assertSelect({"a"}, "SELECT a FROM tmp");
+  assertSelect({"b"}, "SELECT b FROM tmp");
+  assertSelect({"a", "b"}, "SELECT a, b FROM tmp");
+  assertSelect({"b", "a"}, "SELECT b, a FROM tmp");
 
-  auto rowType = ROW({"a", "b"}, {BIGINT(), DOUBLE()});
-  auto plan = PlanBuilder().tableScan(rowType).planNode();
-
-  assertQuery(plan, {split}, "SELECT * FROM tmp");
-
-  // Add a filter on "a".
-  auto filters =
-      common::test::singleSubfieldFilter("a", common::test::lessThan(3));
-
-  plan = PlanBuilder()
-             .tableScan(
-                 rowType,
-                 makeTableHandle(std::move(filters)),
-                 allRegularColumns(rowType))
-             .planNode();
+  // With filters
+  assertSelectWithFilter(
+      {"a"},
+      common::test::singleSubfieldFilter("a", common::test::lessThan(3)),
+      "SELECT a FROM tmp WHERE a < 3");
+  assertSelectWithFilter(
+      {"b"},
+      common::test::singleSubfieldFilter("a", common::test::lessThan(3)),
+      "SELECT b FROM tmp WHERE a < 3");
+  assertSelectWithFilter(
+      {"a", "b"},
+      common::test::singleSubfieldFilter("a", common::test::lessThan(3)),
+      "SELECT a, b FROM tmp WHERE a < 3");
+  assertSelectWithFilter(
+      {"b", "a"},
+      common::test::singleSubfieldFilter("a", common::test::lessThan(3)),
+      "SELECT b, a FROM tmp WHERE a < 3");
+  assertSelectWithFilter(
+      {"b"},
+      common::test::singleSubfieldFilter("a", common::test::lessThan(0)),
+      "SELECT b FROM tmp WHERE a < 0");
 
-  assertQuery(plan, {split}, "SELECT * FROM tmp WHERE a < 3");
+  // TODO: Add filter on b after double filters are supported
```

Comment:
This support is now merged.

Reply:
I just tried to add tests, but the query still doesn't work. See https://github.com/facebookincubator/velox/issues/902

Reply:
The filter should be created as 
```
auto filters =
      common::test::singleSubfieldFilter("b", common::test::lessThanDouble(3.0));
```